src/map/map.ts

```diff
@@ -284,20 +249,26 @@ export class PropertiesMap {
      * @param guid the GUID of the new active viewer
      */
     public setActive(guid: GUID) {
+
         if (this._active !== undefined) {
             const oldData = this._selected.get(this._active);
             assert(oldData !== undefined);
-            oldData.marker.classList.toggle('chsp-active-structure', false);
+            oldData.deactivate();
         }
 
         this._active = guid;
         const data = this._selected.get(this._active);
         assert(data !== undefined);
-        data.marker.classList.toggle('chsp-active-structure', true);
+        data.activate();
 
         if (this._is3D()) {
             this._restyle({'marker.size': this._sizes(1)} as Data, 1);
         }
+
+        const activeData = this._selected.get(guid);
+        assert(activeData !== undefined);
+        const activeIndexes = this._indexer.from_environment(activeData.current);
+        this.activeChanged(guid, activeIndexes);
```

Comment:
why are you calling `activeChanged` here? 

The goal for this callback is to be called when the user change the active marker by interacting with the map; while the goal of `setActive` is to change the active marker on the map when the user interacted with another panel.